sdk/compiler/damlc/daml-opts/daml-opts-types/DA/Daml/Options/Types.hs

```diff
@@ -128,8 +128,7 @@ data Options = Options
   -- packages from remote ledgers.
   , optAllowLargeTuples :: AllowLargeTuples
   -- ^ Do not warn when tuples of size > 5 are used
-  , optWarnBadInterfaceInstances :: WarnBadInterfaceInstances
-  -- ^ Warn for bad interface instances instead of erroring out
+  , optUpgradeInfo :: UpgradeInfo
```

Comment:
All upgrade info is eventually consolidated into this struct.
- `--warn-bad-interface-instances` flag, parsed via DA.Daml.Options
- `--enable-typechecking` flag, parsed via DA.Daml.Options
- `upgrades:` daml.yaml field parsed via DA.Daml.Package.Config, and eventually pushed into `Options` via `updateUpgradePath`